### ISSUE TITLE
json.path.read returns bad JSON object from JSON array of objects

### ISSUE THREAD

**Issue by rkathey** [RPTools/maptool#775]
**Describe the bug**
When you have JSON objects that make up a JSON Array ala [{ },{ }] retrieving an object strips the quotes from the name/values. Example
`	[h:aspects=json.path.read(json,k+".PowerAspects")]
	[h:broadcast(aspects)]
	[h:broadcast(json.type(aspects))]
	[h:aspect=json.path.read(json,k+".PowerAspects.[0]")]
	[h:broadcast(aspect)]
	[h:broadcast(json.type(aspect))]
	[h:json=json.path.set(json,k+".PowerAspects.[0].DisplayKey",first)]
	[h:aspect=json.path.read(json,k+".PowerAspects.[0]")]
	[h:broadcast(aspect)]
	[h:broadcast(json.type(aspect))]` 

Produces the following output, not the lack of quotes around the returned JSON object names and values.

`[{"DisplayKey":"Boost/Lower Trait","Summary":"Increases or decreases a skill or attribute.","DisplaySummary":"Increases or decreases a skill or attribute.","Rank":"Novice","PowerPoints":2,"Range":"Smarts","AoEFlag":0,"Duration":5,"DurationUnits":"Rounds","DurationRounds":5,"Damage":"","ResistTrait":"","PowerModifiers":[]}]
ARRAY
{DisplayKey=Boost/Lower Trait, Summary=Increases or decreases a skill or attribute., DisplaySummary=Increases or decreases a skill or attribute., Rank=Novice, PowerPoints=2, Range=Smarts, AoEFlag=0, Duration=5, DurationUnits=Rounds, DurationRounds=5, Damage=, ResistTrait=, PowerModifiers=[]}
UNKNOWN
{DisplayKey=Boost Trait, Summary=Increases or decreases a skill or attribute., DisplaySummary=Increases or decreases a skill or attribute., Rank=Novice, PowerPoints=2, Range=Smarts, AoEFlag=0, Duration=5, DurationUnits=Rounds, DurationRounds=5, Damage=, ResistTrait=, PowerModifiers=[]}
UNKNOWN`

**To Reproduce**
1. Create a JSON array of JSON objects
2. perform a json.type() on the array and note the returned value is ARRAY
3. perform a json.path.read() to get the first element as shown above
4. display the element returned, it has no quotes around the names or values of the object
5. perform a json.type() on the returned element and confirm it is UNKNOWN

**Expected behavior**
The results of using json.path.read to return an array element should return a valid JSON object if that is what is stored in the array

**Screenshots**

**MapTool Info**
- Version: 1.5.6
- Install: Upgrade 1.5.5

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Version 18.04

**Additional context**

**Comment by Merudo** [comment]
What is `k` in your example?

I think the problem is probably related to my fix to this: https://github.com/RPTools/maptool/issues/612#issuecomment-531581513

There got to be a way to keep the parenthesis intact.

**Comment by Merudo** [comment]
I don't think I can reproduce your issue.

I use the following `json`:

```
{"Bolt": {
    "DisplayKey": "Bolt",
    "Rank": "Novice",
    "PowerAspects": [
      {
        "DisplayKey": "Bolt",
        "Summary": "2d6 ranged attack.",
        "DisplaySummary": "2d6 ranged attack.",
        "Rank": "Novice",
        "PowerPoints": 1,
        "Range": "Smarts",
        "AoEFlag": 0,
        "Duration": 1,
        "DurationUnits": "Rounds",
        "DurationRounds": 1,
        "Damage": "",
        "ResistTrait": "",
        "PowerModifiers": []
      }
    ]
  },
  "BoostLowerTrait": {
    "DisplayKey": "Boost/Lower Trait",
    "Rank": "Novice",
    "PowerAspects": [
      {
        "DisplayKey": "Boost/Lower Trait",
        "Summary": "Increases or decreases a skill or attribute.",
        "DisplaySummary": "Increases or decreases a skill or attribute.",
        "Rank": "Novice",
        "PowerPoints": 2,
        "Range": "Smarts",
        "AoEFlag": 0,
        "Duration": 5,
        "DurationUnits": "Rounds",
        "DurationRounds": 5,
        "Damage": "",
        "ResistTrait": "",
        "PowerModifiers": []
      }
    ]
  }}
```

with the following code:

```

**Comment by rkathey** [comment]
I'll send you a PM with a link to the full object.

**Comment by Merudo** [comment]
Could you make a shorter example that works?

**Comment by rkathey** [comment]
yes. It will be a bit.

**Comment by Phergus** [comment]
Here is a simple code snippet that shows the issue.
```
[h: m = '[{"name":"Troll","HP":75,"Attacks":["Claw","Claw","Bite"]},{"name":"Orc","HP":13,"Attacks":["Sword","Punch"]}]']
[r: json.path.read(m, "[0]")]
```
Output:
``` {name=Troll, HP=75, Attacks=["Claw","Claw","Bite"]}```

**Comment by Merudo** [comment]
Thanks for the example.

In `jayway`, indefinite paths always returns a list instead of a json string. I'll have to figure something out.

**Comment by Azhrei** [comment]
Why are there even equals signs instead of colons?  Shouldn't the element returned be a JSON Object that can be further processed as such?

**Comment by Phergus** [comment]
@Merudo  Is there a different way to express the path that does work?

**Comment by Merudo** [comment]
I think switching `json` provider will work, the default one behaves strangely.

**Comment by Merudo** [comment]
PR #776 should fix this and hopefully not cause any more issues.

The code from @Phergus now returns 

`{"name":"Troll","HP":75,"Attacks":["Claw","Claw","Bite"]}`

The examples on the json.path.read wiki page also give the same results.

**Comment by Phergus** [comment]
Working as expected now.

**Comment by Phergus** [comment]
Oops.  Law of Unintended Consequences strikes again.  String elements are being returned with quotes wrapped around them.
```
[h:orc = json.set("{}", "name", "Orc", "HP", 13, "Attacks", json.append("Sword", "Punch"))]
Path Read: [r: json.path.read( monsters, "Attacks.[1]")]<br>
JSON Get: [r: atck = json.get(json.get(orc,"Attacks"),1)]
```
Returns:
```
Path Read: "Punch"
JSON Get: Punch
```

**Comment by Phergus** [comment]
Fix may be held till 1.5.8 release.
